.buildkite/pipeline.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python3
+import yaml
+import os
+from dataclasses import dataclass, field
+import subprocess
+import fnmatch
+
+from jinja2 import Template
+from pathlib import Path
+
+
+@dataclass()
+class Pipeline:
+    """Buildkite Pipeline object"""
+    groups: list[str]
+
+    def create_entity(self):
+        data = """
+steps:
+{% for group in pipeline.groups -%}
+{{ group.create_entity() }}
+{% endfor -%}
+"""
+
+        tm = Template(data)
+        msg = tm.render(pipeline=self)
+        return msg
```

Comment:
As suggested by @alexsapran below, I think we could simplify the code and get away from Jinja2 templating to native Python data structures. I think they make the code easier to read, allow the IDE to do syntax highlighting and color coding, and more importantly we don't have to pass around strings, but instead render the actual dict as valid yaml text only once.

It was rather straightfoward to do the change, rather than adding noise to this PR review with comments left and right I added the working rewrite in this gist: https://gist.github.com/dliappis/d176ee7667ae9800797c306ac8249db4